packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -211,7 +211,7 @@ export class Table extends Resource {
   /**
    * @attribute
    */
-  public readonly tableStreamArn: string;
+  public readonly tableStreamArn: string | undefined;
```

Comment:
Indicate that this will be undefined if streams are not enabled. 

I suspect this is a breaking change, so I think you’ll have to revert this and do with adding another property called “stream: boolean” (to match the prop name). Feel free to add a note in the v2.0 issue to fix.

Reply:
indicate where, in the jsdoc?

i originally went with the boolean approach, but while that implementation may be backwards compatible, it is fundamentally incorrect IMO. if in production, CloudFormation will effectively cause that value to be undefined, it should be explicitly type-checked against to prevent other areas of the CDK from depending on a value that may not be there and cause unintended behavior (and tons of wasted time).

it is only a "breaking" change, if people have been doing this incorrectly in the past, which in reality, means cloudformation never works, so while the API technically is breaking, no one should have been using the API that way in the first place.

Reply:
You are correct. Let’s stay with your approach. You will need to add a breaking change exclusion line in the root of the repo